### DISCUSSION TITLE
Running Slint via WSL2

### DISCUSSION THREAD

**Discussion by byte2pixel** [slint-ui/slint#4340] [Q&A]
I'm new to Rust and Slint as well so this may be the issue here, not knowing where to configure or find the info I am looking for.

I work on a Windows desktop but I typically develop in WSL2 Ubuntu.  I am able to launch other GUI applications like gedit and xcalc without issue.

All I have done is 
```
cargo generate --git https://github.com/slint-ui/slint-rust-template
cargo build
cargo run
```
Everything seems to look fine as far as the build and the source looks good in neovim but when i run i get this error:

```
❯ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.13s
     Running `target/debug/slint-test`
Error: Other("Could not initialize backend.
Error from Winit backend: Error initializing winit event loop: os error at /home/.../.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.9/src/platform_impl/linux/wayland/event_loop/mod.rs:81: The wayland library could not be loaded
No backends configured.")
```

I have looked around the docs and other issues but I am missing something obvious I'm sure.  Since I can run other linux GUI apps like gedit and x11-apps I am sure there is just something I need to configure correctly to run the Slint rust template. I am trying to figure out how to.. I guess configure the backend so it works. 

Let me know if this is possible or what I can try to configure or what docs can point me to the right direction.

**Comment by byte2pixel** [comment]
I actually was able to get it working.  Updated my gpu drivers in windows, update ubuntu.  updated windows and rebooted.

Now I get this:

The app window and contents do not align though.  I don't see this issue in gedit or other linux apps.  I'm getting close though.

![image](https://github.com/slint-ui/slint/assets/31562317/12795901-b585-47df-b7a0-e78e983dc29c)

**Reply by byte2pixel** [reply]
I found if I hit tab (to select the button) then the window snaps into the frame correctly.

My cursor also disappears when over the app window which makes it difficult to interact with the application.  Any ideas how to fix that?

**Comment by hunger** [comment]
Oh, that looks strange.

I have seen similar on wayland before: Are you sure you are running in X11? Check whether you have a WAYLAND_DISPLAY defined in the terminal you run the Slint UI in and try to unset that if it is set.

**Reply by byte2pixel** [reply]
There is one defined.  I will test this after work today.

```
❯ echo $WAYLAND_DISPLAY
wayland-0
```

**Reply by byte2pixel** [reply]
I tried to unset WAYLAND_DISPLAY and got the following

```
❯ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.15s
     Running `target/debug/slint-test`
thread 'main' panicked at 'Library libxkbcommon-x11.so could not be loaded.', /home/---/.cargo/registry/src/index.crates.io-6f17d22bba15001f/xkbcommon-dl-0.4.1/src/x11.rs:59:28
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I set it back to `wayland-0` and it runs again.  I'm not having issues with the cursor disappearing anymore.  I am guessing that was something else. Now the window and the contents are just not lined up until I hit tab to select the button.

I will probably just bit the bullet and use vscode on windows and try to at least get the vim plugin setup so I can use motions.  Or find something else to work with.

**Reply by hunger** [reply]
HAve you tried installing libxkbcommon-y11? It should be in the package manager of your distro, no need to build anything.

**Reply by byte2pixel** [reply]
I tried it helped some.
Usually starts like this:
![image](https://github.com/slint-ui/slint/assets/31562317/06f4ebf7-71b7-47ba-8ae6-6e5720c8ed12)

I can see the cursor when I am over the window and interact with the minimize, maximize, close and borders.  The contents of the window show up as soon as adjust the window size or cause something to happen like hit tab to select the button.

If I click in the hollow space it interacts with whatever is behind the app window 😆  anyway.  I don't think it is worth figuring out since this is through WSL2 (WSLg)

**Reply by hunger** [reply]
Oh, it is definitely worth figuring this out! I just have no good idea how to actually do that :-(
